bitnami/zookeeper/templates/statefulset.yaml

```diff
@@ -137,13 +143,14 @@ spec:
               value: {{ .Values.maxSessionTimeout | quote }}
             - name: ZOO_SERVERS
               {{- $replicaCount := int .Values.replicaCount }}
+              {{- $minServerId := int .Values.minServerId }}
               {{- $followerPort := int .Values.service.followerPort }}
               {{- $electionPort := int .Values.service.electionPort }}
               {{- $releaseNamespace := include "zookeeper.namespace" . }}
               {{- $zookeeperFullname := include "zookeeper.fullname" . }}
               {{- $zookeeperHeadlessServiceName := printf "%s-%s" $zookeeperFullname "headless" | trunc 63  }}
               {{- $clusterDomain := .Values.clusterDomain }}
-              value: {{ range $i, $e := until $replicaCount }}{{ $zookeeperFullname }}-{{ $e }}.{{ $zookeeperHeadlessServiceName }}.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $followerPort }}:{{ $electionPort }} {{ end }}
+              value: {{ range $i, $e := until $replicaCount }}{{ $zookeeperFullname }}-{{ $e }}.{{ $zookeeperHeadlessServiceName }}.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $followerPort }}:{{ $electionPort }}::{{ add $e $minServerId }} {{ end }}
```

Comment:
This will be conflictive with the workaround to respect existing ids, right? I mean.. This will assign an id based on `{{ .Values.minServerId }}` regardless there was a previous set at `/bitnami/zookeeper/data/myid`. Therefore, the conflict.
Wouldn't be easier to set `ZOO_SERVERS` without including the id?

Reply:
@juan131 
> Wouldn't be easier to set ZOO_SERVERS without including the id?

This unfortunately doesn't work, if `ZOO_SERVERS` are specified without IDs, and IDs are non-standard (0,1,2,..), then  an exception is thrown: 
```
java.lang.RuntimeException: My id 5 not in the peer list	at org.apache.zookeeper.server.quorum.QuorumPeer.start(QuorumPeer.java:1076)	at org.apache.zookeeper.server.quorum.QuorumPeerMain.runFromConfig(QuorumPeerMain.java:227)	at org.apache.zookeeper.server.quorum.QuorumPeerMain.initializeAndRun(QuorumPeerMain.java:136)	at org.apache.zookeeper.server.quorum.QuorumPeerMain.main(QuorumPeerMain.java:90)2020-12-15 11:05:37,258 [myid:5] - ERROR [main:ServiceUtils@42] - Exiting JVM with code 1
```
Alternatively, one could override non-standard `ZOO_SERVERS` via custom `zoo.cfg` or introduce another custom variable which will override standard `ZOO_SERVERS`, This will be anyway required for a stretched cluster scenario to include zookeeper instances from outside the current helm release.

Reply:
You can try to define/export `ZOO_SERVERS` in the same script where `ZOO_SERVER_ID` is being defined and apply similar logic to the one you already implemented.
If you do so, you can remove it from the "env" section.

Reply:
@juan131 setting `ZOO_SERVERS` variable while respecting `myid` from an existing PV will require querying not only local `myid` but also `myid` values of other instances in a StatefulSet, which may be completely uncorrelated to the local one. I don't see an obvious way to do it. To overcome the limitation one can  set `ZOO_SERVERS` variable manually, if `myids` from existing PVs are known and must be preserved.

Reply:
You're right.. We'll have to implement dirty workarounds to fix this... I agree with you, let's keep this simple